python/lsst/ip/isr/overscan.py

```diff
@@ -0,0 +1,504 @@
+    def integerConvert(image):
+        """Return an integer version of the input image.
+
+        Parameters
+        ----------
+        image : `numpy.ndarray`, `lsst.afw.image.Image` or `MaskedImage`
+            Image to convert to integers.
+
+        Returns
+        -------
+        outI : `numpy.ndarray`, `lsst.afw.image.Image` or `MaskedImage`
+            The integer converted image.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if the input image could not be converted.
+        """
+        if hasattr(image, "image"):
+            # Is a maskedImage:
+            imageI = image.image.convertI()
+            outI = afwImage.MaskedImageI(imageI, image.mask, image.variance)
+        elif hasattr(image, "convertI"):
+            # Is an Image:
+            outI = image.convertI()
+        elif hasattr(image, "astype"):
+            # Is a numpy array:
+            outI = image.astype(int)
+        else:
+            raise RuntimeError("Could not convert this to integers: %s %s %s\n",
+                               image, type(image), dir(image))
+        return outI
+
+    # Constant methods
+    def measureConstantOverscan(self, image):
+        """Measure a constant overscan value.
+
+        Parameters
+        ----------
+        image : `lsst.afw.image.Image` or `lsst.afw.image.MaskedImage`
+            Image data to measure the overscan from.
+
+        Returns
+        -------
+        results : `lsst.pipe.base.Struct`
+            Overscan result with entries:
+            - ``overscanValue``: Overscan value to subtract (`float`)
+            - ``maskArray``: Placeholder for a mask array (`list`)
+            - ``isTransposed``: Orientation of the overscan (`bool`)
+        """
+        if self.config.fitType == 'MEDIAN':
+            calcImage = self.integerConvert(image)
+        else:
+            calcImage = image
+
+        fitType = afwMath.stringToStatisticsProperty(self.config.fitType)
+        overscanValue = afwMath.makeStatistics(calcImage, fitType, self.statControl).getValue()
+
+        return pipeBase.Struct(overscanValue=overscanValue,
+                               maskArray=None,
+                               isTransposed=False)
+
+    # Vector correction utilities
+    @staticmethod
+    def getImageArray(image):
+        """Extract the numpy array from the input image.
+
+        Parameters
+        ----------
+        image : `lsst.afw.image.Image` or `lsst.afw.image.MaskedImage`
+            Image data to pull array from.
+
+        calcImage : `numpy.ndarray`
+            Image data array for numpy operating.
+        """
+        if hasattr(image, "getImage"):
+            calcImage = image.getImage().getArray()
+        else:
+            calcImage = image.getArray()
+        return calcImage
+
+    @staticmethod
+    def transpose(image):
+        """Transpose input numpy array if necessary.
+
+        Parameters
+        ----------
+        image : `numpy.ndarray`
```

Comment:
I'm not sure it's a good idea to call a 2D numpy array an `image`, for fear of confusing the types. I usually use `array`.